cellpack/autopack/Environment.py

```diff
@@ -477,6 +298,82 @@ def setSeed(self, seedNum):
+        compartment = self.create_compartment(compartment_key)
+        compartment_info = composition_dict[compartment_key]
+        for region_name, obj_keys in compartment_info.get(
+            "regions", {}
+        ).items():  # check if entry in compositions has regions
+            recipe = Recipe(name=f"{compartment_key}_{region_name}")
+            for key_or_dict in obj_keys:
+                if not isinstance(key_or_dict, dict):
+                    key = key_or_dict
+                    if key in self.compartment_keys:
+                        # deal with compartment
+                        return self._step_down(key)
+                    else:
+                        composition_info = composition_dict[key]
+                        ingredient_info = self._prep_ingredient_info(
+                            composition_info, key
+                        )
+                        self.create_ingredient(recipe, ingredient_info)
+                else:
+                    composition_info = key_or_dict
+                    ingredient_info = self._prep_ingredient_info(composition_info)
+                    self.create_ingredient(recipe, ingredient_info)
+            if region_name == "surface":
+                compartment.setSurfaceRecipe(recipe)
+            elif region_name == "interior":
+                compartment.setInnerRecipe(recipe)
+
+    def create_objects(self):
+        """
+        Instantiate compartments and ingredients contained within the recipe data.
+        """
+        composition_dict = self.recipe_data["composition"]
+
+        if self.root_compartment is not None:
+            # create cytoplasme and set as exterior recipe
+            root_compartment = composition_dict[self.root_compartment]
+            # self.create_compartment(self.root_compartment)
+            external_recipe = Recipe()
+            for region_name, obj_keys in root_compartment.get(
+                "regions", {}
+            ).items():  # check if entry in compositions has regions
+                for key_or_dict in obj_keys:
+                    if not isinstance(key_or_dict, dict):
+
+                        key = key_or_dict
+                        if key in self.compartment_keys:
+                            return self._step_down(key)
+
+                        else:
+                            composition_info = composition_dict[key]
+                            ingredient_info = self._prep_ingredient_info(
+                                composition_info, key
+                            )
+                            self.create_ingredient(external_recipe, **ingredient_info)
+                    else:
+                        composition_info = key_or_dict
+                        ingredient_info = self._prep_ingredient_info(composition_info)
+                        self.create_ingredient(external_recipe, **ingredient_info)
+            self.setExteriorRecipe(external_recipe)
```

Comment:
This code is similar to the definition of `_step_down`, can we have one function that covers both? Perhaps we can deal with the exterior recipe when `_step_down` is called from the root instead of inside this similar code, and only do all of this in one place?

Reply:
If that is substantial enough, we can save it for a future more focused PR to untangle the cytoplasm assumption

Reply:
same as above: https://github.com/mesoscope/cellpack/issues/68